operator/builtin/transformer/add/add.go

```diff
@@ -0,0 +1,104 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package add
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/antonmedv/expr"
+	"github.com/antonmedv/expr/vm"
+	"github.com/open-telemetry/opentelemetry-log-collection/entry"
+	"github.com/open-telemetry/opentelemetry-log-collection/operator"
+	"github.com/open-telemetry/opentelemetry-log-collection/operator/helper"
+)
+
+func init() {
+	operator.Register("add", func() operator.Builder { return NewAddOperatorConfig("") })
+}
+
+// NewAddOperatorConfig creates a new restructure operator config with default values
+func NewAddOperatorConfig(operatorID string) *AddOperatorConfig {
+	return &AddOperatorConfig{
+		TransformerConfig: helper.NewTransformerConfig(operatorID, "add"),
+	}
+}
+
+// AddOperatorConfig is the configuration of a restructure operator
+type AddOperatorConfig struct {
+	helper.TransformerConfig `mapstructure:",squash" yaml:",inline"`
+	Field                    entry.Field `json:"field" yaml:"field"`
+	Value                    interface{} `json:"value,omitempty" yaml:"value,omitempty"`
+}
+
+func (c AddOperatorConfig) Build(context operator.BuildContext) ([]operator.Operator, error) {
+	transformerOperator, err := c.TransformerConfig.Build(context)
+	if err != nil {
+		return nil, err
+	}
+
+	addOperator := &AddOperator{
+		TransformerOperator: transformerOperator,
+		Field:               c.Field,
+	}
+
+	if reflect.TypeOf(c.Value) == reflect.TypeOf(" ") && strings.Contains(c.Value.(string), "EXPR(") {
+		exprStr := strings.Replace(c.Value.(string), "EXPR(", ``, 1)
+		if last := len(exprStr) - 1; last >= 0 && exprStr[last] == ')' {
+			exprStr = exprStr[:last]
+		}
+		compiled, err := expr.Compile(exprStr, expr.AllowUndefinedVariables())
+		if err != nil {
+			return []operator.Operator{addOperator}, fmt.Errorf("failed to compile expression '%s': %w", c.IfExpr, err)
```

Comment:
Typically return `nil, error` when there is a failure like this.

In other words, the operator isn't really built in a useful way, so it's no good.